### PR TITLE
move driver create test to the flutter_tools package

### PR DESCRIPTION
**Pull request by devoncarew**
@yjbanov, move a test of `create --with-driver-test` to the flutter_tools package.

### CODE REVIEW

File: travis/test.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ set -ex
 export PATH="$PWD/bin:$PATH"
 
 # analyze all the Dart code in the repo
-flutter analyze --flutter-repo --no-current-directory --no-current-package --congratulate
+flutter analyze --flutter-repo
```
Comment (eseidelGoogle) [inline travis/test.sh]: Did you mean to remove all these args?

Comment (Hixie) [inline travis/test.sh]: ?

**Comment by Hixie** [comment]
LGTM

File: travis/test.sh (hunk 1 above)
Comment (devoncarew) [inline travis/test.sh]: Yeah, `--congratulate ` is already on by default, and `--no-current-directory --no-current-package` won't have any effect when run from the root of the flutter repo.

File: packages/flutter_tools/test/create_test.dart [hunk 2]
```diff
@@ -70,8 +55,32 @@ void main() {
...
+
+Future<Null> _createAndAnalyzeProject(Directory dir, List<String> createArgs) async {
+  ArtifactStore.flutterRoot = '../..';
+  CreateCommand command = new CreateCommand();
+  CommandRunner runner = new CommandRunner('test_flutter', '')
+    ..addCommand(command);
+  List<String> args = <String>['create'];
+  args.addAll(createArgs);
+  args.add(dir.path);
+  await runner.run(args).then((int code) => expect(code, equals(0)));
```
Comment (yjbanov) [inline packages/flutter_tools/test/create_test.dart]: `=>` implies a value is returned, but `expect` is `void`

Comment (devoncarew) [inline packages/flutter_tools/test/create_test.dart]: I changed this to `int code = await ...`.

**Pull request merged**